track.go

```diff
@@ -223,41 +223,43 @@ func (track *baseTrack) bind(ctx webrtc.TrackLocalContext, specializedTrack Trac
 	keyFrameController, ok := encodedReader.Controller().(codec.KeyFrameController)
 	if ok {
 		stopRead = make(chan struct{})
-		readerBuffer := make([]byte, rtcpInboundMTU)
-		go func() {
-			reader := ctx.RTCPReader()
-			defer func() {
-				track.Close()
-			}()
-			for {
-				select {
-				case <-stopRead:
-					return
-				default:
-				}
+		go track.rtcpReadLoop(ctx.RTCPReader(), keyFrameController, stopRead)
+	}
 
-				_, _, err := reader.Read(readerBuffer, interceptor.Attributes{})
-				if err != nil {
-					track.onError(err)
-					return
-				}
+	return selectedCodec, nil
+}
+
+func (track *baseTrack) rtcpReadLoop(reader interceptor.RTCPReader, keyFrameController codec.KeyFrameController, stopRead chan struct{}) {
+	readerBuffer := make([]byte, rtcpInboundMTU)
+	for {
+		select {
+		case <-stopRead:
+			return
+		default:
+		}
 
-				pkts, err := rtcp.Unmarshal(readerBuffer)
+		readLength, _, err := reader.Read(readerBuffer, interceptor.Attributes{})
+		if err != nil {
+			track.onError(err)
+			return
+		}
+
+		pkts, err := rtcp.Unmarshal(readerBuffer[:readLength])
+		if err != nil {
+			track.onError(err)
+			return
```

Comment:
I think it's better to continue the read loop on RTCP packet processing error.
As RTCP is on UDP, sender won't know receiver stopped reading and will continue sending packets.

Reply:
I agree. I should also probably not exit the read loop if the ForceKeyFrame failed.

Perhaps adding a log to avoid silently discarding a potential problem ?